### PR TITLE
Hide service comments (#52)

### LINKED ISSUE #52: Hide service comments
Service comments like `//readonly` and `//[START section1]` should be hidden in the editor.

### PR DESCRIPTION
**Pull request by alexeyinkin**
Resolves #52

### CODE REVIEW

File: example/lib/custom_code_box.dart [hunk 1]
```diff
@@ -55,16 +56,20 @@ class _CustomCodeBoxState extends State<CustomCodeBox> {
...
+    'a11y-dark',
+    'an-old-hope',
+    'vs2015',
+    'vs',
+    'atom-one-dark',
   ];
 
-  Widget buildDropdown(Iterable<String?> choices, String value, IconData icon,
-      Function(String?) onChanged) {
+  Widget buildDropdown(
```
Comment (nausharipov) [inline example/lib/custom_code_box.dart:67]: Why aren't reset, languageList, buildDropdown, etc. private?

File: example/lib/custom_code_box.dart [hunk 2]
```diff
@@ -108,29 +113,33 @@ class _CustomCodeBoxState extends State<CustomCodeBox> {
...
+          const Spacer(),
           themeDropdown,
-          Spacer(),
+          const Spacer(),
           resetButton,
         ],
       ),
     );
+
     final codeField = InnerField(
```
Comment (nausharipov) [inline example/lib/custom_code_box.dart:131]: Isn't declaring variables in the build method a bad practice? 

https://stackoverflow.com/questions/71095761/declare-variables-in-build-method-or-in-class-itself-flutter

File: example/lib/custom_code_box.dart [hunk 3]
```diff
@@ -168,31 +187,36 @@ class _InnerFieldState extends State<InnerField> {
 
   @override
   Widget build(BuildContext context) {
+    final theme = themes[widget.theme];
+
     return Container(
-      color: _codeController!.theme!['root']!.backgroundColor,
+      color: theme?['root']?.backgroundColor,
```
Comment (nausharipov) [inline example/lib/custom_code_box.dart:193]: Is there a null-safe way to get the theme?

File: example/linux/main.cc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#include "my_application.h"
+
+int main(int argc, char** argv) {
```
Comment (nausharipov) [inline example/linux/main.cc:3]: What is 'example/linux/' for?

File: lib/src/code/code.dart [hunk 5]
```diff
@@ -193,18 +269,104 @@ class Code {
...
+    return isReadOnlyInLineRange(
+      TextRange(
+        start: characterIndexToLineIndex(startChar),
+        end: characterIndexToLineIndex(endChar),
+      ),
+    );
+  }
 
-    for (int line = startLine; line <= endLine; line++) {
+  /// Returns weather any of the lines of this range is read-only.
```
Comment (nausharipov) [inline lib/src/code/code.dart:287]: Typo: 'weather'

File: lib/src/code_field/code_controller.dart [hunk 6]
```diff
@@ -498,7 +498,12 @@ class CodeController extends TextEditingController {
 
     // Return parsing
     if (_language != null) {
-      return _processLanguage(text, CodeTheme.of(context), style);
+      return const SpanBuilder().build(
+        code: _lastCode,
+        theme: _getTheme(context),
+        textStyle: style,
+      );
+      //return _processLanguage(text, CodeTheme.of(context), style);
```
Comment (nausharipov) [inline lib/src/code_field/code_controller.dart:506]: Is this commented code needed?

File: test/src/code/code_get_edit_result.dart [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+  }
+  
+  
+  void method() {
+  }
+  
+}
+''';
+
+void main() {
```
Comment (nausharipov) [inline test/src/code/code_get_edit_result.dart:38]: Is this file missing the "_test" suffix?

File: example/lib/custom_code_box.dart (hunk 1 above)
Comment (alexeyinkin) [reply example/lib/custom_code_box.dart:67]: I only fixed linter issues. This will be done later.

File: example/lib/custom_code_box.dart (hunk 2 above)
Comment (alexeyinkin) [reply example/lib/custom_code_box.dart:131]: This whole example app is one bad practice.

File: example/lib/custom_code_box.dart (hunk 3 above)
Comment (alexeyinkin) [reply example/lib/custom_code_box.dart:193]: What does look null-unsafe here?

File: example/linux/main.cc (hunk 4 above)
Comment (alexeyinkin) [reply example/linux/main.cc:3]: To debug the example app as a linux desktop app. It has a better debugger.

File: lib/src/code/code.dart (hunk 5 above)
Comment (alexeyinkin) [reply lib/src/code/code.dart:287]: +

File: lib/src/code_field/code_controller.dart (hunk 6 above)
Comment (alexeyinkin) [reply lib/src/code_field/code_controller.dart:506]: We may need to compare output to the old renderer if any highlighting bugs show up.

File: example/lib/custom_code_box.dart (hunk 3 above)
Comment (nausharipov) [reply example/lib/custom_code_box.dart:193]: If there is no corresponding theme, the Container is left with a (presumably undesired) null color.

**Review by nausharipov — APPROVED** [review]

**Pull request merged**